biotrainer/embedders/embedding_service.py

```diff
@@ -230,6 +232,42 @@ def _save_and_reset_embeddings(self, embeddings: Dict[str, ndarray], last_save_i
         del embeddings
         return last_save_id, {}
 
+    @staticmethod
+    def reduce_embeddings_dimension(
+        embeddings: Dict[str, Any],
+        dimension_reduction_method: str,
+        reduced_dimension: int) -> Dict[str, Any]:
+        """
+        Reduces the dimension of per-residue embeddings or per-protein embeddings
+
+        Parameters:
+            embeddings (Dict[str, ndarray]): Dictionary of embeddings.
+            embedder: The embedder used for reducing embeddings.
+
+        Returns:
+            out (Dict[str, ndarray]): Dictionary of reduced embeddings.
+        """
+        sorted_keys = sorted(list(embeddings.keys()))
```

Comment:
Please add a logging statement for start like: 
`logger.info(f"Starting embeddings dimensionality reduction via method {dimension_reduction_method}")`